src/libpriv/rpmostree-core.c

```diff
@@ -2297,14 +2295,21 @@ rpmostree_context_relabel (RpmOstreeContext *self,
                                      G_CALLBACK (on_hifstate_percentage_changed),
                                      prefix);
 
+  /* Prep a txn and tmpdir for all of the relabels */
+  if (!ostree_repo_prepare_transaction (ostreerepo, NULL, cancellable, error))
+    return FALSE;
+  if (!rpmostree_context_ensure_tmpdir (self, "relabel", error))
+    return FALSE;
+
   guint n_changed_files = 0;
   guint n_changed_pkgs = 0;
   const guint n_to_relabel = self->pkgs_to_relabel->len;
+
   for (guint i = 0; i < n_to_relabel; i++)
     {
         DnfPackage *pkg = self->pkgs_to_relabel->pdata[i];
         guint pkg_n_changed = 0;
-        if (!relabel_one_package (ostreerepo, pkg, self->sepolicy,
+        if (!relabel_one_package (self, ostreerepo, pkg, self->sepolicy,
                                   &pkg_n_changed, cancellable, error))
           return FALSE;
```

Comment:
Don't we also need to call `abort_transaction` before returning here?

Reply:
Yep, good catch.  Factored out https://github.com/projectatomic/rpm-ostree/pull/972